### PR TITLE
[Feature] (Multi-Catalog) support query hll column in doris jdbc table - part 1

### LINKED ISSUE #17895: [Feature] (Doris External Table) support HLL/BITMAP for Doris External Table with JDBC
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently,  when we query a Doris table with JDBC external table,  the HLL/BITMAP column is not supported.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by GoGoWen**
# Proposed changes

Issue Number: close #17895

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
Hi @GoGoWen , there is a schedule for this task, such as how many steps?
And need to add regression test for this.
You can refer to `regression-test/suites/jdbc_catalog_p0/test_mysql_jdbc_catalog.groovy`

**Comment by GoGoWen** [comment]
> Hi @GoGoWen , there is a schedule for this task, such as how many steps? And need to add regression test for this. You can refer to `regression-test/suites/jdbc_catalog_p0/test_mysql_jdbc_catalog.groovy`

Hi  @morningman
1, "how many steps" -> now i am working on part 2 - support Bitmap,  we also want to query Doris Bitmap column from another doris cluster.   
2  "And need to add regression test for this"  -> i will add the regression test.

File: be/src/vec/exec/vjdbc_connector.cpp [hunk 1]
```diff
@@ -648,6 +724,8 @@ Status JdbcConnector::_register_func_id(JNIEnv* env) {
 
     RETURN_IF_ERROR(register_id(_executor_clazz, "getBlock", JDBC_EXECUTOR_GET_BLOCK_SIGNATURE,
                                 _executor_get_blocks_id));
+    RETURN_IF_ERROR(register_id(_executor_clazz, "getBlock", JDBC_EXECUTOR_GET_BLOCK_NEW_SIGNATURE,
```
Comment (morningman) [inline be/src/vec/exec/vjdbc_connector.cpp:727]: `JDBC_EXECUTOR_GET_BLOCK_NEW_SIGNATURE`, may be we can use a more meaningful name?

File: fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java [hunk 2]
```diff
@@ -194,6 +194,29 @@ public void rollbackTrans() throws UdfRuntimeException {
...
+    public List<Object[]> getBlock(int batchSize, Object colsArray) throws UdfRuntimeException {
+        try {
+            ArrayList<Integer> colsTypes = (ArrayList<Integer>) colsArray;
+            Integer[] colArray = new Integer[colsTypes.size()];
+            colArray = colsTypes.toArray(colArray);
+            int columnCount = resultSetMetaData.getColumnCount();
+            curBlockRows = 0;
+            do {
+                for (int i = 0; i < columnCount; ++i) {
+                    if (colArray[i] > 0) {
```
Comment (morningman) [inline fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java:206]: Add comment to explain the value saved in `colArray`. Like `> 0` means this is a hll type.

**Comment by morningman** [comment]
Also need to update the col mapping doc in `https://doris.apache.org/zh-CN/docs/dev/lakehouse/multi-catalog/jdbc#doris`

File: be/src/vec/data_types/data_type_hll.cpp [hunk 3]
```diff
@@ -126,4 +126,30 @@ void DataTypeHLL::to_string(const class doris::vectorized::IColumn& column, size
...
+
+Status DataTypeHLL::from_string(ReadBuffer& rb, IColumn* column) const {
+    auto& data_column = assert_cast<ColumnHLL&>(*column);
+    auto& data = data_column.get_data();
+
+    std::string str;
+    str = rb.to_string();
+    HyperLogLog hll;
+    if (!hll.deserialize(Slice(str))) {
+        LOG(WARNING) << "deserialize hll from string fail!";
```
[the remark this answers is not in the document]
Comment (GoGoWen) [reply be/src/vec/data_types/data_type_hll.cpp:149]: in previous design, if some error hll row deserialize fail can not effect other correct hll.    And  now seems  we make it fail fast should be better.

File: be/src/vec/exec/vjdbc_connector.cpp (hunk 1 above)
Comment (GoGoWen) [reply be/src/vec/exec/vjdbc_connector.cpp:727]: will change to JDBC_EXECUTOR_GET_BLOCK_WITH_TYPES_SIGNATURE, is that OK?

File: regression-test/suites/jdbc_catalog_p0/test_doris_jdbc_catalog.groovy [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        PARTITION p202308 VALUES [('2023-08-01'), ('2023-09-01')),
+        PARTITION p202309 VALUES [('2023-09-01'), ('2023-10-01')))
+        DISTRIBUTED BY HASH(`pin_id`) BUCKETS 16
+        PROPERTIES (
+        "replication_allocation" = "tag.location.default: 1",
+        "dynamic_partition.enable" = "true",
+        "dynamic_partition.time_unit" = "MONTH",
+        "dynamic_partition.time_zone" = "Asia/Shanghai",
+        "dynamic_partition.start" = "-2147483648",
+        "dynamic_partition.end" = "5",
```
[the remark this answers is not in the document]
Comment (GoGoWen) [reply regression-test/suites/jdbc_catalog_p0/test_doris_jdbc_catalog.groovy]: Done, removed dynamic partition

File: fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java (hunk 2 above)
Comment (GoGoWen) [reply fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java:218]: Done

**Comment by GoGoWen** [comment]
> Also need to update the col mapping doc in `https://doris.apache.org/zh-CN/docs/dev/lakehouse/multi-catalog/jdbc#doris`

Done

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by GoGoWen** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.11 seconds
 stream load tsv:          432 seconds loaded 74807831229 Bytes, about 165 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          60 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          76.2 seconds inserted 10000000 Rows, about 131K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230521025646_clickbench_pr_147783.html

**Comment by GoGoWen** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.96 seconds
 stream load tsv:          428 seconds loaded 74807831229 Bytes, about 166 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          78.7 seconds inserted 10000000 Rows, about 127K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230522051830_clickbench_pr_148090.html

**Comment by GoGoWen** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.38 seconds
 stream load tsv:          435 seconds loaded 74807831229 Bytes, about 164 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          79.2 seconds inserted 10000000 Rows, about 126K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230522070622_clickbench_pr_148156.html

**Comment by GoGoWen** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 37.11 seconds
 stream load tsv:          427 seconds loaded 74807831229 Bytes, about 167 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          82.0 seconds inserted 10000000 Rows, about 121K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230522090232_clickbench_pr_148228.html

**Comment by GoGoWen** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.93 seconds
 stream load tsv:          430 seconds loaded 74807831229 Bytes, about 165 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          79.0 seconds inserted 10000000 Rows, about 126K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230522142734_clickbench_pr_148506.html

**Comment by GoGoWen** [comment]
run p0

**Comment by GoGoWen** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**